src/hammer-tech/hammer_tech.py

```diff
@@ -384,6 +413,84 @@ def tech_defined_libraries(self) -> List[Library]:
         """
         return list(self.config.libraries)
 
+    def get_extra_macro_sizes(self) -> List[MacroSize]:
+        """
+        Get the list of extra macro sizes from the config.
+        See vlsi.technology.extra_macro_sizes in defaults.yml.
+        :return: List of extra macro sizes.
+        """
+        if not self.has_setting("vlsi.technology.extra_macro_sizes"):
+            # If the key doesn't exist we can safely say there are none.
+            return []
+
+        extra_macro_sizes = self.get_setting("vlsi.technology.extra_macro_sizes")
+        if not isinstance(extra_macro_sizes, list):
+            raise ValueError("extra_macro_sizes was not a list")
+        else:
+            return list(map(MacroSize.from_setting, extra_macro_sizes))
+
+    def get_tech_macro_sizes(self) -> List[MacroSize]:
+        """
+        Compile a list of all macros which have size information, using LEF files.
+        This also considers any extra IP libraries.
+        :return: List of all macros' size information.
+        """
+        extra_libraries = self.get_extra_libraries()  # type: List[ExtraLibrary]
+
+        # Enhance the LEF file to also extract the name of the library
```

Comment:
I don't understand this comment. It doesn't appear to be about the function defined below it.

Reply:
Oops, there is a typo. Will fix.